src/extensions/advanced-image/index.js

```diff
@@ -0,0 +1,138 @@
+import AttachmentCropControl from '../../components/image-crop-control/attachment-crop-control';
+
+import { assign } from 'lodash';
+const { createHigherOrderComponent } = wp.compose;
+const { Fragment } = wp.element;
+const { InspectorControls } = wp.editor;
+const { PanelBody } = wp.components;
+const { addFilter } = wp.hooks;
+const { __ } = wp.i18n;
+
+const supportedBlocks = [
+	'core/image',
+	'core/cover',
+];
+
+const addPositioningControl = ( settings, name ) => {
+	if ( ! supportedBlocks.includes( name ) ) {
+		return settings;
+	}
+
+	settings.attributes = assign( settings.attributes, {
+		cropX: {
+			type: 'number',
+			default: 0,
+		},
+		cropY: {
+			type: 'number',
+			default: 0,
+		},
+		cropWidth: {
+			type: 'number',
+			default: 100,
+		},
+		cropHeight: {
+			type: 'number',
+			default: 100,
+		},
+		cropRotation: {
+			type: 'number',
+			default: 0,
+		},
+	} );
+
+	return settings;
+};
+
+const updateDebounce = _.debounce( function( callback ) {
+	callback();
+}, 1500 );
+
+const positioningControl = createHigherOrderComponent( ( BlockEdit ) => {
+	return ( props ) => {
+		// Do nothing if it's another block than our defined ones.
+		if ( ! supportedBlocks.includes( props.name ) ) {
+			return (
+				<BlockEdit { ...props } />
+			);
+		}
+
+		const {
+			attributes,
+			setAttributes,
+		} = props;
+
+		const { cropX, cropY, cropWidth, cropHeight, cropRotation } = attributes;
+		let currentAttributes = _.extend( {}, attributes );
+
+		// Only Gallery images supported at this time
+		if ( ! currentAttributes.id ) {
+			return (
+				<BlockEdit { ...props } />
+			);
+		}
+
+		const updateImage = function() {
+			jQuery.post( global.ajaxurl, {
```

Comment:
Changed `ajaxurl` to `global.ajaxurl`. This is to avoid ESLint warnings.

Reply:
For future reference, we can also throw `/* global ajaxurl */` at the top of the file to mark it as global in ESLint.